src/utils/issue.ts

```diff
@@ -15,7 +15,7 @@ export async function getAssignedIssues(context: Context, username: string): Pro
   try {
     return await context.octokit
       .paginate(context.octokit.search.issuesAndPullRequests, {
-        q: `org:${payload.repository.owner.login} assignee:${username} is:open is:issue`,
+        q: `org:ubiquity org:ubiquity-os-marketplace org:ubiquity-os assignee:${username} is:open is:issue`,
```

Comment:
I think the most robust way to get all "within network" is to get the list of orgs that installed the bot

Reply:
Addressed

Reply:
Afaik this isn't feasible @0x4007 because we need the kernel' private key which has been spoken about across a few issues but started [here](https://github.com/ubiquity-os-marketplace/command-start-stop/pull/35)

Reply:
I would like to take a step back.
Can you provide some insights why we would like to search assigned issues within the network? I can't think of the reason why searching for issues within the same organization is not enough?
It won't be possible to list all orgs that installed the bot without using app private key.

Reply:
> Can you provide some insights why we would like to search assigned issues within the network?

To get a better picture of how over-assigned the contributor might be across multiple partners (eventually we'll have loads).

---

1. Global storage would enable us to track a userbase across all partners. The user object can be updated any time they are assigned an issue within the network and we can track with ease. This can serve us in hundreds of ways not just with this use-case.
2. Parse devpool json (better with all issues filled, theres a task for it) + search API for `username: pull_requests`.
3. Depending on the user' visibility settings you'll likely be able to do it with GraphQL API but I'm not as knowledgeable with that API, I use the official [GitHub GQL Explorer](https://docs.github.com/en/graphql/overview/explorer) to investigate when I need to.

Reply:
I see, understood.

Based on your suggestion to use Github Search API, I played with it a bit and I think it is possible to search for issues where the bot has interacted with, using this query: `is:open is:issue commenter:ubiquity-os[bot] assignee:<user>`..
I think it can work, and we don't need to use the private key anymore. 

The question now is should I check for `ubiquity-os[bot]` explicitly or make it configurable in `.ubiquity-os.config.yml`.

Reply:
Whichever way you feel is most effective dude go for it.

https://github.com/ubiquity-os-marketplace/command-start-stop/blob/1ceebb363cf414b5a9ad00c101ea5e1d706424fb/src/types/env.ts#L9

Use this variable to find out the bot's name instead of a new config item but if not then I guess you'd need a new config item if using that approach but I feel the right thing to do is use it's ID vs a new config item.

Reply:
Unfortunately cannot search for App with an `id`, it needs to be a `node_id`. And we don't have it,.

Reply:
> Unfortunately cannot search for App with an `id`, it needs to be a `node_id`. And we don't have it,.

ohh I was hoping that wasn't the case. You could use `author: <xx> is:pr` and do an org comparison that way instead otherwise @0x4007 or @gentlementlegen will approve the new config item.

Reply:
With this approach, the idea is to compare with devpools.json right? Checking the json file, but it seems https://raw.githubusercontent.com/ubiquity/devpool-directory/refs/heads/development/devpool-issues.json is not available anymore?

And a question: shouldn't it be enough to just build the list of orgs using devpools.json? Is it useful to compare with the results from `author:<xxx> is:pr` query?

Reply:
https://github.com/ubiquity/devpool-directory/tree/__STORAGE__ - storage has it's own branch.

> And a question: shouldn't it be enough to just build the list of orgs using devpools.json? Is it useful to compare with the results from author:<xxx> is:pr query?

This fn is trying to find assigned issues so

- using search API to get all PRs, then filter, then get the linked issue via GQL or PR body parsing
- using search API with `assignee:keyrxng is:issue`, then just filter is another but it doesn't take PRs into consideration (not sure if that's something we should consider in this context)

option two is probs best.

Reply:
> Use this variable to find out the bot's name instead of a new config item but if not then I guess you'd need a new config item if using that approach but I feel the right thing to do is use it's ID vs a new config item.

Actually the kernel/SDK exports the app id and user ID of the production bot. I was thinking that maybe this would be actually sent along to the plugin because it can change if you run your own kernel

> Pretty sure the REST api does not provide a method to search for a user with their user.id or node_id so you'd need to use the GQL API.

This is the api: https://docs.github.com/en/rest/users/users?apiVersion=2022-11-28#get-a-user-using-their-id

Reply:
> Afaik this isn't feasible @0x4007 because we need the kernel' private key which has been spoken about across a few issues. It was ofc the telegram bot most recently [ubiquity-os-marketplace/ubiquity-os-kernel-telegram#2 (comment)](https://github.com/ubiquity-os-marketplace/ubiquity-os-kernel-telegram/pull/2#discussion_r1766516363)

Try this. Its available in the org secrets.

https://github.com/ubiquity/cloudflare-deploy-action/blob/3e07491a4785469af9eee58bf697203428c0f8ee/action.yml#L50-L56

Reply:
Hmm.. So we use this

> > Afaik this isn't feasible @0x4007 because we need the kernel' private key which has been spoken about across a few issues. It was ofc the telegram bot most recently [ubiquity-os-marketplace/ubiquity-os-kernel-telegram#2 (comment)](https://github.com/ubiquity-os-marketplace/ubiquity-os-kernel-telegram/pull/2#discussion_r1766516363)
> 
> Try this. `APP_ID` and `APP_PRIVATE_KEY` have now been added to the @ubiquity-os-marketplace org secrets.
> 
> GET: https://github.com/ubiquity/cloudflare-deploy-action/blob/3e07491a4785469af9eee58bf697203428c0f8ee/action.yml#L50-L56
> 
> SET: https://github.com/ubiquity/cloudflare-deploy-action/blob/3e07491a4785469af9eee58bf697203428c0f8ee/action.yml#L62
> 
> Just make sure to get from secrets instead of it being "passed in" like it is here.

Hmm, I assume we can only access secrets in Github/ workflow. So, I imagine calling `listInstallations` inside a Github Workflow. I am not sure how to pass this information to the plugin..??